lib/spack/spack/util/web.py

```diff
@@ -577,106 +566,30 @@ def find_versions_of_archive(archive_urls, list_url=None, list_depth=0):
-    capitalized; whether words are separated; or, if separated, whether they
-    are so by a dash (-), underscore (_), or space ( ).  Header names that
-    cannot be mapped as described above are returned unaltered.
-
-    For example: The standard spelling of "Content-length" would be substituted
-    for any of the following names:
-    - Content-length
-    - content_length
-    - contentlength
-    - content_Length
-    - contentLength
-    - content Length
-
-    ... and any other header name, such as "Content-encoding", would not be
-    altered, regardless of spelling.
-
-    If headers is a string, then it (or an appropriate substitute) is returned.
-
-    If headers is a non-empty tuple, headers[0] is a string, and there exists a
-    standardized spelling for header[0] that differs from it, then a new tuple
-    is returned.  This tuple has the same elements as headers, except the first
-    element is the standardized spelling for headers[0].
-
-    If headers is a sequence, then a new list is considered, where each element
-    is its corresponding element in headers, but mapped as above if a string or
-    tuple.  This new list is returned if at least one of its elements differ
-    from their corrsponding element in headers.
-
-    If headers is a mapping, then a new dict is considered, where the key in
-    each item is the key of its corresponding item in headers, mapped as above
-    if a string or tuple.  The value is taken from the corresponding item.  If
-    the keys of multiple items in headers map to the same key after being
-    standardized, then the value for the resulting item is undefined.  The new
-    dict is returned if at least one of its items has a key that differs from
-    that of their corresponding item in headers, or if the keys of multiple
-    items in headers map to the same key after being standardized.
-
-    In all other cases headers is returned unaltered.
-    """
-    if isinstance(headers, six.string_types):
-        for standardized_spelling, other_spellings in (
-                HTTP_HEADER_NAME_ALIASES.items()):
-            if headers in other_spellings:
-                if headers == standardized_spelling:
-                    return headers
-                return standardized_spelling
-        return headers
-
-    if isinstance(headers, tuple):
-        if not headers:
-            return headers
-        old = headers[0]
-        if isinstance(old, six.string_types):
-            new = standardize_header_names(old)
-            if old is not new:
-                return (new,) + headers[1:]
-        return headers
-
-    try:
-        changed = False
-        new_dict = {}
-        for key, value in headers.items():
-            if isinstance(key, (tuple, six.string_types)):
-                old_key, key = key, standardize_header_names(key)
-                changed = changed or key is not old_key
+def get_header(headers, header_name):
```

Comment:
Nice work condensing this! There are a few places in this file it should be applied as well. Specifically in the in the `read_from_url()` method on lines 152 and 164. I had to make that modification to get this working in my setup.

Reply:
@wylderkeane thanks for the catch! 👍